config.js

```diff
@@ -1,21 +1,25 @@
 var redis = require('redis');
 
-exports.config = {
-  redis_server_url: "127.0.0.1",
-  redis_server_port: 6379,
+var config = {
+  redis_server_host: process.env.REDIS_SERVER_HOST || "127.0.0.1",
+  redis_server_port: process.env.REDIS_SERVER_PORT || 6379,
   redis_server_requires_auth: false,
-  redis_server_password: ""
+  redis_server_password: process.env.REDIS_SERVER_PASSWORD || "",
 };
 
 exports.getRedisClient = function() {
-  if (this.config.redis_server_url && this.config.redis_server_port) {
+  if (config.redis_server_host && config.redis_server_port) {
     var options = {};
 
-    if (this.config.redis_server_requires_auth && this.config.redis_server_password) {
```

Comment:
I'd remove the `requires_auth` variable and just look for a password. Then you can greatly simplify this code